openssl-sys/src/openssl_shim.c

```diff
@@ -86,3 +92,295 @@ long SSL_set_tlsext_host_name_shim(SSL *s, char *name) {
 STACK_OF(X509_EXTENSION) *X509_get_extensions_shim(X509 *x) {
     return x->cert_info ? x->cert_info->extensions : NULL;
 }
+
+#if OPENSSL_VERSION_NUMBER < 0x1000200L
```

Comment:
This seems like a bit much to add here - it seems like it would make more sense to put this stuff under a feature flag like we do for other things that require modern versions e.g. DTLS.